### DISCUSSION TITLE
HyperHDR 20b1: HDR toggle not working for Flatbuffers (webos)

### DISCUSSION THREAD

**Discussion by Xploder** [awawa-dev/HyperHDR#693] [General]
Hey,
trying HyperHDR 20 b1 with WebOS Flatbuffers and it seems that the Global HDR toggle doesn't work anymore for it, so automatic HDR switching is not working anymore. Is there something I'm missing?

Thanks!

**Comment by awawa-dev** [comment]
Hi
HDR toggle button works on default for USB grabbers. If you built a version without software/hardware grabbers (default build configuration for webOS I think, don't maintain it) then the toggle should work for flatbuffers/protobuffer instead.

**Comment by Xploder** [comment]
Hey Awawa,
Thanks for replying!
I've been trying to contact you via Discord again first but it seems like we're not part of the same server anymore so it didn't work :smile: 

I'm not running it on my TV but on my Homeserver (Unraid) via Docker.
Any way you see how I could get back this functionality there?

Thanks!

**Comment by awawa-dev** [comment]
I think the fastest way to achieve the result you want is to build it with: -DENABLE_FRAMEBUFFER=OFF -DENABLE_PIPEWIRE=OFF -DENABLE_X11=OFF -DENABLE_V4L2=OFF You can even use Github Action to build it. The next few weeks are for collecting feedback without developing, unless something really critical comes up.

**Reply by Xploder** [reply]
Okay I will have to find another host platform and build it myself then, as I was just using a docker image from dockerhub.
Are there benefits to changing the behavior? It worked flawlessly for me with Piccap automatically switching this toggle through the API.

**Reply by awawa-dev** [reply]
Do you need any of USB grabbers support in your docker? So why not use Github Action to build it: fork the repo, **enable** Github Action for it first, and paste the parameters here: https://github.com/awawa-dev/HyperHDR/blob/558c7e03d7da0d52653cd2fe68d96e3634fae227/.ci/ci_build.sh#L122 take about 1 minute to do it and about 30-60 minutes of waiting till Github finishes it.

**Reply by awawa-dev** [reply]
Sorry I forgot about CEC that you must disable: -DENABLE_CEC=OFF
I will remember to check later, if the toggle can work also when the grabbers are enabled in the build configuration.

**Reply by Xploder** [reply]
After compiling my own version with mentioned flags (without USB support), this works flawlessly again.
My docker image is available under xploder270/hyperhdr:latest for anyone interested.

Thank you!

EDIT: It worked without the CEC flag

**Comment by awawa-dev** [comment]
Added fix in https://github.com/awawa-dev/HyperHDR/pull/705 Note: that is not needed if HyperHDR runs directly on WebOS
